core/store/src/trie/from_flat.rs

```diff
@@ -0,0 +1,61 @@
+use crate::flat::{store_helper, FlatStorageManager};
+use crate::{ShardTries, Store, Trie, TrieConfig, TrieDBStorage, TrieStorage};
+use near_primitives::{shard_layout::ShardUId, state::FlatStateValue};
+use std::time::Instant;
+
+pub fn construct_trie_from_flat(store: Store, write_store: Store, shard_uid: ShardUId) {
+    let timer = Instant::now();
+    let trie_storage = TrieDBStorage::new(store.clone(), shard_uid);
+
+    let mut entries =
+        store_helper::iter_flat_state_entries(shard_uid, &store, None, None).map(|entry| {
+            let (key, value) = entry.unwrap();
+            let value = match value {
+                FlatStateValue::Ref(ref_value) => {
+                    trie_storage.retrieve_raw_bytes(&ref_value.hash).unwrap().to_vec()
```

Comment:
Does Ref exist anymore? I remember there was a migration. I assume it doesn't since otherwise it would be much slower to do random lookups from the trie storage.

Reply:
I think it still does right? I remember we were using inlined values only for values smaller than 4 KB? Maybe @Longarithm can confirm.